src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsSyncService.kt

```diff
@@ -106,6 +109,21 @@ class ContactPersonProfileDetailsSyncService(
     }
   }
 
+  suspend fun readmissionSwitchBooking(event: ReadmissionSwitchBookingEvent) {
+    // We're filtering for this reason in the Topic subscription filter, but this is just in case we listen for other prisoner received events later
+    if (event.additionalInformation.reason != "READMISSION_SWITCH_BOOKING") return
+
+    val prisonerNumber = event.additionalInformation.nomsNumber
+
+    if (eventFeatureSwitch.isEnabled("personal-relationships-api.domestic-status.created")) {
+      syncProfileDetail(prisonerNumber, 0, MARITAL)
+    }
+
+    if (eventFeatureSwitch.isEnabled("personal-relationships-api.number-of-children.created")) {
+      syncProfileDetail(prisonerNumber, 0, CHILD)
+    }
+  }
```

Comment:
The main change is to listen for prisoner received events with reason of switch booking, then sync everything back to NOMIS.